staging/src/k8s.io/apiserver/pkg/endpoints/handlers/workflow_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestWorkflowID(t *testing.T) {
+	tests := []struct {
+		subResource string
+		verb        string
+		contentType string
+		userAgent   string
+		wanted      string
+	}{
+		{
+			verb:   "PUT",
+			wanted: "default-put",
+		},
+		{
+			subResource: "scale",
+			verb:        "PUT",
+			userAgent:   "kubectl/v1.10.4 (linux/amd64) kubernetes/5ca598b",
+			wanted:      "kubectl-scale-put",
+		},
+		{
+			verb:        "PATCH",
+			contentType: "application/json-patch+json",
+			userAgent:   "Random stuff",
+			wanted:      "random stuff-patch",
```

Comment:
Another TODO item for your list: what's the validation rules for a workflow id? Does it allow spaces? (bleh) How long can it be?

Reply:
Yeah, I'll change the userAgent pattern to a regex that just selects the first word or somethnig.